pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/ChannelManager.java

```diff
@@ -36,32 +41,69 @@
  * query/job/stages.
  */
 public class ChannelManager {
+  /**
+   * Map from (hostname, port) to the ManagedChannel with all known channels
+   */
   private final ConcurrentHashMap<Pair<String, Integer>, ManagedChannel> _channelMap = new ConcurrentHashMap<>();
   private final TlsConfig _tlsConfig;
+  /**
+   * The idle timeout for the channel, which cannot be disabled in gRPC.
```

Comment:
Just checking once again - why can't we default to setting the idle timeout to `Long.MAX_VALUE` which would basically disable the idle mode for gRPC channels. We could still allow configuring the idle timeout manually, but this would preclude the need for this pinger mechanism altogether, no? This has even been suggested by maintainers of `grpc-java` [here](https://github.com/grpc/grpc-java/issues/2022#issuecomment-1059294635).